### PR TITLE
Booking models roughcut

### PR DESCRIPTION
**Pull request by nickevansuk**
PR for the purpose of initial review

### CODE REVIEW

File: versions/2.x/models/booking/Offer.json [hunk 1]
```diff
@@ -0,0 +1,202 @@
+{
+   "type":"Offer",
+   "derivedFrom":"https://schema.org/Offer",
+   "isJsonLd":true,
+   "idFormat":"https://schema.org/URL",
+   "sampleId":"https://example.com/Offer/",
+   "status":{
+      "REQUEST":{
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: Sure. The allcaps was out of a dim recollection of Java enums being used in allcaps for state-machine states. But given that everything else is camel-case, using it here makes sense.

File: versions/2.x/models/booking/Offer.json [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+         "options":[
+            "https://openactive.io/Required",
+            "https://openactive.io/Optional",
+            "https://openactive.io/Unavailable"
+         ],
+         "description":"Indicates whether to accept this offer, a participant must book in advance, whether they must pay on attending, or have option to do either. Values must be one of  https://openactive.io/Required,  https://openactive.io/Optional or  https://openactive.io/Unavailable."
+      },
+      "prepayment":{
+         "attributeName":"prepayment",
+         "sameAs":"https://www.openactive.io/ns/#prepayment",
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: Ok, thx. Will fix.

File: versions/2.x/models/booking/Offer.json [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+         "fieldName":"type",
+         "requiredType":"https://schema.org/Text",
+         "requiredContent":"Offer"
+      },
+      "@id":{
+         "fieldName":"@id",
+         "requiredType":"https://schema.org/URL"
+      },
+      "price":{
+         "attributeName":"price",
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: Oops.

File: versions/2.x/models/booking/Offer.json [hunk 4]
```diff
@@ -0,0 +1,202 @@
+{
+   "type":"Offer",
+   "derivedFrom":"https://schema.org/Offer",
+   "isJsonLd":true,
+   "idFormat":"https://schema.org/URL",
+   "sampleId":"https://example.com/Offer/",
+   "status":{
+      "REQUEST":{
+        "hasId":true,
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: Hmmm, but is it always the case that the ID is useful in all states? For example, for a schema:Order, the ID of the original Request doesn't have any real force.

File: versions/2.x/models/booking/Offer.json [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+      "availableDeliveryMethod",
+      "businessFunction",
+      "category",
+      "deliveryLeadTime",
+      "eligibleCustomerType",
+      "eligibleDuration",
+      "eligibleQuantity",
+      "eligibleRegion",
+      "eligibleTransactionVolume",
+      "gtin12",
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: Gotcha.

File: versions/2.x/models/booking/Offer.json [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      "potentialAction",
+      "sameAs",
+      "subjectOf",
+      "url"
+   ],
+   "fields":{
+      "@type":{
+         "fieldName":"type",
+         "requiredType":"https://schema.org/Text",
+         "requiredContent":"Offer"
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Offer.json]: +1

File: versions/2.x/models/booking/OrderProposal.json [hunk 7]
```diff
@@ -0,0 +1,237 @@
...
+      "REQUEST":{
+         "hasId":false,
+         "requiredFields":[
+            "@type",
+            "broker",
+            "brokerRole",
+            "customer",
+            "orderedItem",
+            "totalPaymentDue",
+            "OrderProposalStatus"
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/OrderProposal.json]: +1

File: versions/2.x/models/booking/Order.json [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+      "image",
+      "mainEntityOfPage",
+      "name",
+      "potentialAction",
+      "sameAs",
+      "subjectOf",
+      "url"
+   ],
+   "fields":{
+      "@type":{
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Order.json]: Ah, that trick. Will do.

File: versions/2.x/models/booking/Order.json [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+         "fieldName":"type",
+         "requiredType":"https://schema.org/Text",
+         "requiredContent":"Order"
+      },
+      "@id":{
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/Order.json]: +1

File: versions/2.x/models/booking/OrderProposal.json [hunk 10]
```diff
@@ -0,0 +1,237 @@
...
+         "model":"Organization",
+         "sameAs":"https://schema.org/seller",
+         "description":[
+            "The organisation (schema:Organization) providing access to events or facilities via a Booking System. e.g. a leisure provider running yoga classes."
+         ]
+      },
+      "taxMode":{
+         "fieldName":"taxMode",
+         "sameAs":"https://www.openactive.io/ns/#taxMode",
+         "model":"#TaxMode",
```
[the remark this answers is not in the document]
Comment (thill-odi) [reply versions/2.x/models/booking/OrderProposal.json]: +1

File: versions/2.x/models/booking/Offer.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (nickevansuk) [reply versions/2.x/models/booking/Offer.json]: yep you're right, sorry forgot it also generated examples. ignore this comment in that case

File: versions/2.x/models/booking/Offer.json (hunk 5 above)
[the remark this answers is not in the document]
Comment (nickevansuk) [reply versions/2.x/models/booking/Offer.json]: This still needs addressing

**Comment by nickevansuk** [comment]
Merging into master, with a note to clean up excess notInSpec at a later date

**Pull request merged**